toreview/KadActionsBuilder.java

```diff
@@ -0,0 +1,311 @@
+
+    }
+
+    /**
+     * Default constructor, builder will accept all valid IDs, (as indicated in KadAction)
+     */
+    public KadActionsBuilder() {
+        MAX_ACTION_ID = KadAction.MAX_ID;
+    }
+
+    private boolean isValidID(int id) {
+        return id >= KadAction.MIN_ID && id <= MAX_ACTION_ID;
+    }
+
+    /**
+     * @param peer     the peer that will be pinged
+     * @param actionID the unique ID of the action
+     * @return if the given ID is valid: an action built to execute the ping command to the peer, otherwise returns an INVALID_ACTION
+     */
+    public KadAction buildPing(int actionID, SMSPeer peer) {
+        if (isValidID(actionID))
+            return new KadAction(
+                    peer,
+                    KadAction.ActionType.PING,
+                    actionID,
+                    DEFAULT_PARTS, DEFAULT_PARTS,
+                    KadAction.PayloadType.IGNORED,
+                    PAYLOAD_IGNORED
+            );
+        else return INVALID_ACTION;
+    }
+
+    /**
+     * @param request the ping request received from a peer (that will be the receiver for the response)
+     * @return a response action built to respond to the received ping command from the peer, or an invalid KAD action if the given action wasn't coherent with this building method (actionType)
+     */
+    public KadAction buildPingAnsw(KadAction request) {
+        if (request.getActionType() == KadAction.ActionType.PING)
+            return new KadAction(
+                    request.getPeer(),
+                    KadAction.ActionType.PING_ANSWER,
+                    request.getOperationId(),
+                    DEFAULT_PARTS, DEFAULT_PARTS,
+                    KadAction.PayloadType.IGNORED,
+                    PAYLOAD_IGNORED
+            );
+        else
+            return INVALID_ACTION;
+    }
+
+    /**
+     * @param peer     the receiving peer (the peer to invite)
+     * @param actionID the unique ID of the action
+     * @return if the given ID is valid: an action built to execute the invite command to the peer, otherwise returns an INVALID_ACTION
+     */
+    public KadAction buildInvite(int actionID, SMSPeer peer) {
+        if (isValidID(actionID))
+            return new KadAction(
+                    peer,
+                    KadAction.ActionType.INVITE,
+                    actionID,
+                    DEFAULT_PARTS, DEFAULT_PARTS,
+                    KadAction.PayloadType.IGNORED,
+                    PAYLOAD_IGNORED
+            );
+        else
+            return INVALID_ACTION;
+    }
+
+    /**
+     * @param request  the received invite request
+     * @param accepted set to indicate if the answer accepts the invite or if it rejects it
+     * @return a response action built to respond to the received invite, or an invalid KAD action if the given action wasn't coherent with this building method (actionType)
+     */
+    public KadAction buildInviteAnsw(KadAction request, boolean accepted) {

```

Comment:
A questo punto completa il nome per 2 lettere :)

Reply:
Sì me l'avevano fatto notare già😂. A me piaceva più così ma lo cambierò xD. La "problematica" è che per essere agevoli avevo definito l'interfaccia ActionsBuilder in modo che questi metodi si potessero già "utilizzare" nel resto del codice prima che questa classe fosse pronta. Avevamo messo questi nomi e siamo andati avanti così, poi andare a cambiare era un po' difficoltoso per via di un paio di problemi strutturali che adesso abbiamo più o meno risolto tutti.
Perciò ho lasciato i nomi per adesso, ma in effetti non ha tanto senso come abbreviazione, adesso li posso cambiare

Reply:
CTRL + SHIFT + R per aggiornare un nome su tutto il progetto

Reply:
Non era quello il problema, android studio fa un ottimo refactoring automatico. Il fatto è che la repository era organizzata male, ma adesso abbiamo sistemato.